File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
package Raindrops

fun convert(x: Int): String {
    var result = ""

    if (x % 3 == 0) {
        result += "Pling"
    }

    if (x % 5 == 0) {
        result += "Plang"
    }

    if (x % 7 == 0) {
        result += "Plong"
    }

    return if (result == "") "$x" else result
}

File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

namespace raindrops {

std::string convert(const int x) {
    std::string result;

    if (!(x % 3))
        result += "Pling";

    if (!(x % 5))
        result += "Plang";

    if (!(x % 7))
        result += "Plong";

    return result.empty() ? std::to_string(x) : result;
}

} // namespace raindrops
